src/sql/azdata.proposed.d.ts

```diff
@@ -533,6 +533,39 @@ declare module 'azdata' {
 		type?: ExtensionNodeType;
 	}
 
+	export interface AccountDisplayInfo {
+		email?: string;
+		name?: string;
+	}
+
+	export interface AccountProvider {
+		/**
+		 * Generates a security token for the provided account and tenant
+		 * @param account The account to generate a security token for
+		 * @param resource The resource to get the token for
+		 * @return Promise to return a security token object
+		 */
+		getAccountSecurityToken(account: Account, tenant: string, resource: AzureResource): Thenable<accounts.AccountSecurityToken | undefined>;
+	}
+
+	export interface AccountKey {
+		/**
+		 * A version string for an account
+		 */
+		accountVersion?: string;
+		/**
+		 * Auth Library used to add the account
+		 */
+		authLibrary?: string;
```

Comment:
Why does an account care what library it was added with?

Reply:
This is so we can filter the accounts based on the authLibrary that's currently selected

Reply:
But why do we need to filter the accounts? The account shouldn't care what library it's using - that only changes how we retrieve the token and make calls but it's still all for the same account.

Reply:
the accounts are maintained in two different token caches so we need to be able to select the right one based on the authLibrary
Also, with MSAL the account IDs are a bit different so they're not really interchangeable (I couldn't just keep one account for both libraries, which is what I believe you're suggesting)